README.md

```diff
@@ -1,83 +1,127 @@
 
-```
-lein jlink run
-lein jlink test
-```
+If you are using modules that are not distributed with the JDK, you can download and add them to your project with the `:jlink-modules-paths` keyword in your `project.clj`. For instance, if you downloaded the [JavaFX modules][6] from [Gluon's website][7] (referred to as "jmods"), you would unpack them somewhere on your machine and then add a reference to them like so...
 
-A more flexible way is to create a profile in your project:
+    :jlink-module-paths ["/opt/java/javafx-jmods-14.0.1"]
+    
+If you are running on Windows, you will want to escape the `\` character in your paths with `\\`. For example:
 
-```clj
-:profiles {
-  :jlink {:java-cmd "./target/jlink/bin/java"}
-}
-```
+    :jlink-module-paths ["C:\\Program Files\\Java\\javafx-jmods-14.0.1"]
+    
+External modules _only_ work at compile time, they cannot be passed to your default `java` command at runtime. There are two solutions:
+
++ You could download and install the SDK for the module. This plugin supports this but not all modules provide an SDK.
++ Create a runtime image and use the `java` command from the image to execute your project; this is the recommended solution.
+
+#### Adding a Module SDK
+
+If you happen to have the SDK for a module installed and you don't want to build a custom runtime image, you can add the path to the SDK to your `project.clj` file on with the `:jlink-sdk-paths` key. For example...
+
+    :jlink-sdk-paths ["C:\\Program Files\\Java\\javafx-sdk-14\\lib"]
+    
+That being said, building a custom image might be easier since you don't need to install anything except the modules.
+    
+### Buidling a Custom Runtime Environment
+
+A custom runtime environment will build a new Java runtime for your project, including _only_ the modules your project uses. If you have external modules but lack access to an SDK, you will _need_ to create a custom runtime in order to actually run your project.
+
+First you need to let the plugin know that you want to use a custom runtime when interacting with your project. Set the `:jlink-custom-jre` key in your `project.clj` to `true`.
 
-### Assemble and package
+    :jlink-jre-image true
+    
+This flag also lets the middleware know that it shouldn't use the `java` available in `JAVA_HOME` (or the JDK you specified) but should use the `java` in the custom runtime image instead.
 
-By running `lein jlink assemble`, we create a uberjar and put it into
-custom JRE directory. A launcher script is also generated in
-`target/jlink/bin` that can run your application with this JRE.
+Create a custom Java environment with the plugin's `init` task. It will call out to the `jlink` tool that is bundled with your JDK to create a new "runtime image". This image will be in the "image" directory at the root of your project. If you would like to store the runtime image in a different location, you can provide that location with the `:jlink-jre-image-path` key in your `project.clj` file.
```

Comment:
Note that `:jlink-jre-image-path` should be a relative path to `target`. You cannot set it to your own external JRE.

Reply:
I moved the image directory out of the target directory because Leiningen would keep deleting it during it's `clean` task (Leiningen would clean before creating the uberjar).

My project will build an image for each target platform, I think I'll have a Leiningen profile for each platform, this key let's me change the path for the image so that I can build several of them. 

My project is a work in progress but I have managed to get most of the big pieces working.

* [XMLTool, Java 14 branch](https://github.com/cmiles74/xmltool/tree/java-14-wip)

Reply:
OK. My concern was, since the jlink JRE was reproducible, it might be confusing to put generated content outside `target` while removable via our patched `clean`.

The script under `bin` was designed to benefit docker image generation. The script works as entrypoint. I suggest to keep it and perhaps you can add windows' `bat` version to it.

Reply:
I really tried to make it work in the "target" directory but when the plugin's "assemble" task ran it would create the image and then Leiningen would delete it when it ran its "uberjar" task; I'd end up with the uberjar and the launcher script but no image. 😦 

I moved the Unix-like launcher script back in and added a PowerShell and a batch script. Any of the three could be used as the entry point by Docker, depending on the Docker image.